wicketstuff-restannotations-parent/restannotations/src/main/java/org/wicketstuff/rest/resource/RestResourceLocaleLocator.java

```diff
@@ -0,0 +1,35 @@
+package org.wicketstuff.rest.resource;
+
+import org.apache.wicket.Session;
+
+import java.util.Locale;
+
+/**
+ * A singleton which locates {@link Locale} for {@link AbstractRestResource}
+ */
+public class RestResourceLocaleLocator {
+
+	private static RestResourceLocaleLocator instance;
+
+	private LocaleResolver localeResolver;
+
+	public Locale getLocale() {
+		return localeResolver != null ? localeResolver.getLocale() : Session.get().getLocale();
```

Comment:
This could do an extra effort to prevent creating a Wicket `Session`:
```
if (Session.exists()) {
   locale = Session.get().getLocale();
} else {
   local = RequestCycle.get().getRequest().getLocale();
}
```

Reply:
What you propose is quite different from existing behavior, and could be a breaking change. The purpose of the resolver is that we can supply something like RequestCycle.get().getRequest().getLocale() or even return a static locale based on props or config instead of existing session.

Do you think we want to change the default behavior of how we  get a locale?

Reply:
Essentially this Session.get() is an issue by itself. I implemented this locale locator because a while ago one of our customers decided to start submitting a zillion requests per minute, which caused a zillion sessions, which was not very nice in terms of server resources and not quite restful. I don't remember, but I'm pretty sure tomcat has exploded.

So, maybe we should completely replace this `Session.get()` with something else.

Reply:
> which was not very nice in terms of server resources

`Session.get()` creates a **temporary** *Wicket* Session. Unless you call `session.bind()` there is no HttpSession being created!
So zillion of requests would lead to higher CPU usage due to garbage collection but not really using a lot of resources.

Reply:
As Martin said Session.get() creates a temporary session, so se keep the resource stateless as It should be. There Is however room for improvement. We could introduce a method like 'getLocale' which can be overriden to use a custom logic to retrieve the locale.

Reply:
Thank you! Maybe my resource locator is redundant here then. I can simply write a protected getLocale method with Session.get().getLocale(), implementations can override it and use whatever locator they like. Let me check if there a particular reason for this locator to exist. What do you think?

Reply:
Back to my computer. I had a look, I do not see a particular reason for this `RestResourceLocaleLocator` to exist. If you, colleagues, do not see any value in this class I can remove it from the pull request and replace it with protected `getLocale` method.

Reply:
Ah. The place where it's called is a static method! So, the locator is, probably a solution.